backend/api/battles/serializers.py

```diff
@@ -38,6 +45,21 @@ def validate_creator_pokemon_3(self, value):
             raise serializers.ValidationError("Sorry, this pokemon was not found")
         return Pokemon.objects.get(name=value)
 
+    def validate_opponent_pokemon_1(self, value):
+        if not pokemon_exists(value):
```

Comment:
You are repeating this quite a lot, I think you should have a method that does this validation and just call it from these `validate` methods.